packages/sdk/typescript/human-protocol-sdk/src/escrow.ts

```diff
@@ -0,0 +1,746 @@
+import { EscrowStatus, NetworkData } from './types';
+
+export default class EscrowClient {
+  private escrowFactoryContract: EscrowFactory;
+  private escrowContract?: Escrow;
+  private signerOrProvider: Signer | Provider;
+  public network: NetworkData;
+
+  /**
+   * **Escrow constructor**
+   *
+   *   * @param {IClientParams} clientParams - Init client parameters
+   */
+  constructor(readonly clientParams: IClientParams) {
+    this.network = clientParams.network;
+
+    this.escrowFactoryContract = EscrowFactory__factory.connect(
+      clientParams.network.factoryAddress,
+      clientParams.signerOrProvider
+    );
+    this.signerOrProvider = clientParams.signerOrProvider;
+  }
+
+  /**
+   * Creates an escrow contract that uses the token passed to pay oracle fees and reward workers.
+   *
+   * @param {string} tokenAddress - Token address to use for pay outs.
+   * @param {string[]} trustedHandlers - Array of addresses that can perform actions on the contract.
+   * @returns {Promise<string>} - Return the address of the escrow created.
+   * @throws {Error} - An error object if an error occurred.
+   */
+  public async createEscrow(
+    tokenAddress: string,
+    trustedHandlers: string[]
+  ): Promise<string> {
+    if (!Signer.isSigner(this.signerOrProvider)) {
+      throw ErrorSigner;
+    }
+
+    if (!ethers.utils.isAddress(tokenAddress)) {
+      throw ErrorInvalidTokenAddress;
+    }
+
+    trustedHandlers.forEach((trustedHandler) => {
+      if (!ethers.utils.isAddress(trustedHandler)) {
+        throw ErrorInvalidAddress;
+      }
+    });
+
+    try {
+      await this.escrowFactoryContract.createEscrow(
+        tokenAddress,
+        trustedHandlers
+      );
+
+      return this.escrowFactoryContract.lastEscrow();
+    } catch (e: any) {
+      return throwError(e);
+    }
+  }
+
+  /**
+   * Sets up the parameters of the escrow.
+   *
+   * @param {string} escrowAddress - Address of the escrow to set up.
+   * @param {IEscrowConfig} escrowConfig - Configuration object with escrow settings.
+   * @returns {Promise<void>}
+   * @throws {Error} - An error object if an error occurred.
+   */
+  async setup(
+    escrowAddress: string,
+    escrowConfig: IEscrowConfig
+  ): Promise<void> {
+    const {
+      recordingOracle,
+      reputationOracle,
+      recordingOracleFee,
+      reputationOracleFee,
+      manifestUrl,
+      hash,
+    } = escrowConfig;
+
+    if (!Signer.isSigner(this.signerOrProvider)) {
+      throw ErrorSigner;
+    }
+
+    if (!ethers.utils.isAddress(recordingOracle)) {
+      throw ErrorInvalidRecordingOracleAddressProvided;
+    }
+
+    if (!ethers.utils.isAddress(reputationOracle)) {
+      throw ErrorInvalidReputationOracleAddressProvided;
+    }
+
+    if (!ethers.utils.isAddress(escrowAddress)) {
+      throw ErrorInvalidEscrowAddressProvided;
+    }
+
+    if (!(await this.escrowFactoryContract.hasEscrow(escrowAddress))) {
+      throw ErrorEscrowAddressIsNotProvidedByFactory;
```

Comment:
We can make this call as the last validation to avoid contract calls when another param is wrong. Same for all methods

Reply:
Agree